src/main/java/io/strimzi/kafka/bridge/mqtt/utils/MappingRulesLoader.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.mqtt.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.strimzi.kafka.bridge.mqtt.core.MqttKafkaMapper;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Helper class to Load the rules from the configuration file
+ */
+public class MappingRulesLoader {
+
+    private static final MappingRulesLoader instance = new MappingRulesLoader();
+    // path of the topic mapping rule file
+    private String mapperRuleFilePath;
+
+    /**
+     * Private constructor
+     */
+    private MappingRulesLoader() {
+    }
+
+    /**
+     * Get the singleton instance of the MappingRulesLoader
+     *
+     * @return the singleton instance of the MappingRulesLoader
+     */
+    public static MappingRulesLoader getInstance() {
+        return instance;
+    }
+
+    /**
+     * Set the path of the mapper rule file
+     *
+     * @param mapperRuleFilePath the path of the mapper rule file
+     */
+    public void setMapperRuleFilePath(String mapperRuleFilePath) {
+        this.mapperRuleFilePath = mapperRuleFilePath;
+    }
+
+    /**
+     * Load the mapping rules from the file system and create the mapper instance.
+     *
+     * @see MqttKafkaMapper
+     */
+    public List<MappingRule> loadRules() throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        StringBuilder json = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new FileReader(this.mapperRuleFilePath));
+        String line;
+        // build the JSON string
+        while ((line = reader.readLine()) != null) {
+            json.append(line);
+        }
```

Comment:
can you give it a try to this method? https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/ObjectMapper.html#readValue(java.io.File,%20java.lang.Class)

Instead of reading the file first, line by line, building a String and then readValue on that String, the above method could be able to get a `File` instance directly and loading directly into a list of `MappinRule`(s).

Reply:
I just tried, you can replace everything with just one line:

```java
return mapper.readValue(Path.of(this.mapperRuleFilePath).toFile(), mapper.getTypeFactory().constructCollectionType(List.class, MappingRule.class));
```

Reply:
This is wise! :). Thanks for the suggestion.

Reply:
Should we do something like this for more clarity ?
` // get the file from the path
        final File mappingRulesJsonFile = Path.of(this.mapperRuleFilePath).toFile();
        
        final CollectionType collectionType = mapper.getTypeFactory().constructCollectionType(List.class, MappingRule.class);
        
        // deserialize the JSON array to a list of MappingRule objects
        return mapper.readValue(mappingRulesJsonFile, collectionType);`

Reply:
I don't think so, it's just a bunch of more lines of code and local variables, which don't make it clearer imho.

Reply:
I have fixed that in my recent commit